datatable/java/datatable/src/test/java/io/cucumber/datatable/DataTableTypeRegistryTest.java

```diff
@@ -120,4 +120,24 @@ public void returns_entry_data_table_type() {
 
         assertSame(entry, lookupTableTypeByType);
     }
+
+    @Test
+    public void parse_decimal_with_english_locale() {
+        expectDecimalValueForDecimalStringInLocale(Locale.ENGLISH, 12.45, "12.45");
+    }
+
+    @Test
+    public void parse_decimal_with_polish_locale() {
+        expectDecimalValueForDecimalStringInLocale(Locale.forLanguageTag("pl"), 12.45, "12,45");
+
+    }
+
+    private void expectDecimalValueForDecimalStringInLocale(Locale locale, double expectedDecimalValue, String decimalString) {
```

Comment:
The use of custom assert methods usually indicates that the test is not properly structured. In this instance test is testing a property of the `NumberParser` but does so through the `DataTableTypeRegistry` and needs a fairly complicated setup to do so.

Creating a `NumberParserTest` and moving this test would solve that problem.

Reply:
Yes but it would not then check if DataTableRegistry uses NumberParser properly. We would need have another test in DataTableTypeRegistryTest which checks if proper method from NumberParser is invoked.

Reply:
Fair enough. I still think these test can be much more concise though. To the point of not needing the the custom assert.

Reply:
@lsuski I've pushed a fix that will make the tests more concise.